batdata/streaming.py

```diff
@@ -0,0 +1,85 @@
+"""Retrieve data in smaller chunks from a large HDF5 file"""
+from pathlib import Path
+from contextlib import contextmanager
+from typing import Union, Iterator, Dict, List
+
+import pandas as pd
+from pandas import HDFStore
+from pandas.io.pytables import TableIterator
+
+from batdata.data import BatteryDataset
+
+
+@contextmanager
+def _get_raw_data_iterator_h5(hdf5_path: Union[Path, str, HDFStore], **kwargs) -> Iterator[pd.DataFrame]:
+    """Open an iterator over chunks of an HDF5 file"""
+    chunk_iter: TableIterator = pd.read_hdf(hdf5_path, key='raw_data', iterator=True, **kwargs)
+    try:
+        yield chunk_iter
+    finally:
+        if not isinstance(hdf5_path, HDFStore):
+            chunk_iter.close()
+
+
+def iterate_records_from_file(hdf5_path: Union[Path, str, HDFStore], **kwargs) -> Iterator[Dict[str, Union[str, float, int]]]:
+    """Stream individual records from a file
+
+    Keyword arguments are passed to :meth:`~pandas.read_hdf`.
+
+    Args:
+        hdf5_path: Path to the data file
+
+    Yields:
+        Individual rows from the "raw_data" section of the HDF5 file
+    """
+
+    with _get_raw_data_iterator_h5(hdf5_path, **kwargs) as chunk_iter:
+        for chunk in chunk_iter:
+            for _, row in chunk.iterrows():
+                yield row
+
+
+def iterate_cycles_from_file(hdf5_path: Union[Path, str, HDFStore],
+                             make_dataset: bool = False,
+                             **kwargs) -> Iterator[Union[pd.DataFrame, BatteryDataset]]:
+    """Stream single-cycle datasets from the HDF5 file
+
+    Keyword arguments are passed to :meth:`~pandas.read_hdf`.
+
+    Args:

```

Comment:
What are the `**kwargs` used for? I noticed in the tests that you had a `chunksize` as one of them?

Reply:
They control how the HDF5 reader backing pandas works. Options include changing which columns are read, a query string, etc.